File: 1436-destination-city/1436-destination-city.py
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        cities = set()
        
        for path in paths:
            cities.add(path[0])
        
        for path in paths:
            if path[1] not in cities:
                return path[1]
        
        return ""

File: LeetCode/1436-destination-city/1436-destination-city.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Set<String> cities = new HashSet<String>();
        
        for (List<String> path: paths) {
            cities.add(path.get(0));
        }
        
        for (List<String> path: paths) {
            if (!cities.contains(path.get(1))) {
                return path.get(1);
            }
        }
        
        return "";
    }
}